src/main/java/uk/gov/hmcts/darts/common/repository/TransformedMediaRepository.java

```diff
@@ -58,4 +59,30 @@ AND upper(tod.status.description) <> 'MARKED FOR DELETION'
         """)
     List<TransformedMediaEntity> findAllDeletableTransformedMedia(OffsetDateTime createdAtOrLastAccessedDateTime);
 
-}
+    @Query("""
+        SELECT tm
+            FROM TransformedMediaEntity tm
+            JOIN tm.mediaRequest media
+            JOIN media.hearing hearing
+            JOIN hearing.courtCase courtCase
+            JOIN hearing.courtroom courtroom
+            JOIN courtroom.courthouse courthouse
+        WHERE
+           (:mediaId IS NULL OR (:mediaId IS NOT NULL AND media.id=:mediaId)) AND
+           (:caseNumber IS NULL  OR (:caseNumber IS NOT NULL AND courtCase.caseNumber=:caseNumber)) AND
+           (:courtHouseDisplayName IS NULL OR (:courtHouseDisplayName IS NOT NULL AND courthouse.displayName like CONCAT('%', :courtHouseDisplayName, '%'))) AND
+           (:hearingDate IS NULL OR (:hearingDate IS NOT NULL AND hearing.hearingDate=:hearingDate )) AND
+           (:owner IS NULL OR (:owner IS NOT NULL AND media.currentOwner.userFullName like CONCAT('%', :owner, '%'))) AND
+           (:requestedBy IS NULL OR (:requestedBy IS NOT NULL AND tm.createdBy.userFullName like CONCAT('%', :requestedBy, '%'))) AND
+           ((cast(:requestedAtFrom as TIMESTAMP)) IS NULL OR (:requestedAtFrom IS NOT NULL AND tm.createdDateTime >= :requestedAtFrom)) AND
```

Comment:
@chrisbellingham-hmcts is this the correct database value for the requested at or should I be using the media request table?